Singularity

```diff
@@ -23,67 +23,32 @@ From: centos:7
+    apt-get update
+    apt-get install -y build-essential unzip git wget \
+        fontconfig libbz2-dev liblzma-dev libssl-dev \
+        libffi-dev libsqlite3-dev
 
     echo ===== Installing Python ===== >/dev/null
-    wget -q https://www.python.org/ftp/python/3.8.3/Python-3.8.3.tar.xz
-    tar xJf Python*
-    rm Python*.xz
-    cd Python*
-    ./configure --enable-optimizations
-    make altinstall
-    cd ..
-    rm -rf Python*
-    ln -s /usr/local/bin/python3.8 /usr/local/bin/python3
+    apt-get install -y python3 python3-pip
 
     echo ===== Installing blast ===== >/dev/null
-    cd /root
-    # Saved our own copy, because download was slow from ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.6.0/ncbi-blast-2.6.0+-1.x86_64.rpm
-    wget https://github.com/cfe-lab/MiCall/releases/download/v7.12.dev28/ncbi-blast-2.6.0+-1.x86_64.rpm
-    yum install -q -y ncbi-blast-2.6.0+-1.x86_64.rpm
-    rm ncbi-blast-2.6.0+-1.x86_64.rpm
+    apt-get install -y ncbi-blast+
 
     echo ===== Installing Python packages ===== >/dev/null
-    wget -q https://bootstrap.pypa.io/get-pip.py
-    python3 get-pip.py
-    rm get-pip.py
     cd /opt/primer_finder
-    pip install .
-
-    # Note that yum installs need to happen before the alternatives command is run, yum will not work with Python3
-    echo ===== Installing R ===== >/dev/null
-    # Shouldn't need this since it was installed earlier
-    # yum install epel-release -y
-    yum install R -y
-    Rscript /opt/primer_finder/gene_splicer/configure_r.sh 2> /dev/null
+    pip3 install .
 
     echo ===== Installing minimap2 ===== >/dev/null
-    cd /opt
-    wget https://github.com/lh3/minimap2/releases/download/v2.17/minimap2-2.17_x64-linux.tar.bz2
-    tar -jxvf ./minimap2-2.17_x64-linux.tar.bz2
-    ln -s /opt/minimap2-2.17_x64-linux/minimap2 /bin
-    rm minimap2-2.17_x64-linux.tar.bz2
+    apt-get install -y minimap2
 
-    echo ===== Installing HIVSeqinR ===== >/dev/null
+    echo ===== Installing hivseqinr ===== >/dev/null
+    apt-get install -y libz-dev libcurl4-openssl-dev libxml2-dev
+    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y r-base
+    Rscript /opt/primer_finder/gene_splicer/configure_r.sh
     python3 -m gene_splicer.hivseqinr /opt/hivseqinr
 
     # Clean up
-    yum --setopt=groupremove_leaf_only=1 groupremove -q -y 'development tools'
-    yum remove -q -y epel-release wget git
-    yum autoremove -q -y
-    yum clean all
-    rm -rf /var/cache/yum
-
-    ## CAUTION! This changes the default python command to python3!
-    ## This breaks many things, including yum!
-    ## To switch back to python2, use this command:
-    # sudo alternatives --set python /usr/bin/python2
-    alternatives --install /usr/bin/python python /usr/bin/python2 50
-    alternatives --install /usr/bin/python python /usr/local/bin/python3 60
+    apt-get remove -y wget git build-essential
```

Comment:
Try also running `apt-get clean` and `rm -rf /var/lib/apt/lists/*` to see if they reduce the image size.

Reply:
Reduced by 30MB (about 7%)